crates/ironrdp-client-blocking/src/rdp.rs

```diff
@@ -0,0 +1,270 @@
+use crate::config::Config;
+use ironrdp::connector::{ConnectionResult, ConnectorResult};
+use ironrdp::graphics::image_processing::PixelFormat;
+use ironrdp::pdu::input::fast_path::FastPathInputEvent;
+use ironrdp::session::image::DecodedImage;
+use ironrdp::session::{ActiveStage, ActiveStageOutput, SessionResult};
+use ironrdp::{connector, session};
+use smallvec::SmallVec;
+use sspi::network_client::reqwest_network_client::RequestClientFactory;
+use std::io;
+use std::net::TcpStream;
+use std::sync::Arc;
+use winit::event_loop::EventLoopProxy;
+
+#[derive(Debug)]
+pub enum RdpOutputEvent {
+    Image { buffer: Vec<u32>, width: u16, height: u16 },
+    ConnectionFailure(connector::ConnectorError),
+    Terminated(SessionResult<()>),
+}
+
+#[derive(Debug)]
+pub enum RdpInputEvent {
+    Resize {
+        width: u16,
+        height: u16,
+    },
+    FastPath(SmallVec<[FastPathInputEvent; 2]>),
+    Close,
+    /// DevNull is a special event that is used to check if the input event channel is closed.
+    DevNull,
```

Comment:
This is a hack, but I believe does the trick of replacing

https://github.com/Devolutions/IronRDP/blob/b742a3faee30f0ca7a1c259dcedbfc14af14768c/crates/ironrdp-client/src/gui.rs#L236-L238

since `crossbeam::channel::Sender`s don't have a corresponding `is_closed` method.

Reply:
Sounds good!